robots/preservation/update_moab.rb

```diff
@@ -0,0 +1,25 @@
+
+module Preservation
+  # Robot either creates or updates moab version then places it into the Moab directory.
+  class UpdateMoab < Base
+    ROBOT_NAME = 'update-moab'.freeze
+
+    def initialize(opts={})
+      super(REPOSITORY, WORKFLOW_NAME, ROBOT_NAME, opts)
+    end
+
+    def perform(druid)
+      storage_object = Moab::StorageServices.find_storage_object(druid, true)
+      storage_object.object_pathname.mkpath
+      update_moab(storage_object)
+    end
+
+    def update_moab(storage_object)
```

Comment:
i made a point of only having "perform" method be public and all the specific methods for THIS robot as private.   I won't go to the mat over it;  it would mean a slight rewriting of your tests (they would call perform instead of update_moab)